### DISCUSSION TITLE
SimpliSafe login failed with error: Error: No authentication credentials detected

### DISCUSSION THREAD

**Discussion by warrengottlieb** [homebridge-simplisafe3/homebridge-simplisafe3#311] [Q&A] [answered]
Hello, looking for some authentication help please: 

I followed the login instructions using `homebridge-simplisafe3 login` and was able to get a "Credentials retrieved successfully." message. When I restart homebridge (and/or the container) I see the message from the logs: 

`SimpliSafe login failed with error: Error: No authentication credentials detected`

My config.json originally had my auth settings but I have removed everything except the platform and name lines. Not sure what else to look for at this point. In sum, my authentication appears successful and then not after reboot.

Thanks!

**Comment by shamoon** [comment, marked as answer]
"container" makes me think youre using docker? The auth info is stored in a file so if you are using docker you need to setup a volume to persist, e.g. 
```
    volumes:
      - /home/docker/homebridge:/homebridge
```

Let me know if that isnt the issue

**Reply by warrengottlieb** [reply]
Hi, thanks for the response. I'm not sure if that is the issue. I am using docker—an oznu/homebridge image on a Synology. I'm newish to docker and especially so running from the NAS, but the storage is persistent, that is, any changes I make are present on reboot.

**Reply by shamoon** [reply]
Ok, does your storage folder contain a file `'simplisafe3auth.json'` ?

**Comment by warrengottlieb** [comment]
Aha! 

I searched for that file and it's present in `/root/.homebridge/simplisafe3auth.json`

My Homebridge folder that contains my `config.json` file is actually just  `/homebridge/`

I copied the auth file into my /homebridge folder and restarted...and it's working! 

Thank you so much.

**Reply by shamoon** [reply]
Thats great. Ah yes, so usually on docker you would get a warning `Error: Supplied path /root/.homebridge does not exist` because docker uses a different location,. Normally youd have to do `homebridge-simplisafe3 login -d /homebridge/`. You must have created that directory for some reason and thus it was (incorrectly) used but the plugin didnt report an issue.

Probably an opportunity to handle this for the user, or at least I'll add that to the docs.

Thanks!
